### PR TITLE
fix: Remove screen with restriction about do not allow user to buy ticket on fallback state

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes https://github.com/AtB-AS/kundevendt/issues/17026

### CODE REVIEW

**Comment by marius-at-atb** [comment]
I think this was the only use case for this screen, so unless there is a plan to use it for something else, we should delete the screen as well.

**Comment by rosvik** [comment]
I think we should still show the screen if fallback isn't active, but the user has not selected phone. E.g. for t:card users.

**Comment by jorelosorio** [comment]
> I think this was the only use case for this screen, so unless there is a plan to use it for something else, we should delete the screen as well.

The screen is still used in other cases, but we can remove the message if is not used in other places.

**Comment by jorelosorio** [comment]
> I think we should still show the screen if fallback isn't active, but the user has not selected phone. E.g. for t:card users.

So the screen is presented, but with the option to skip it? according to the description and what I talked to Sebastian it should work as bus ticket is working, no intermediate screens on it (if something is wrong with mobile token)

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx [hunk 1]
```diff
@@ -72,25 +72,6 @@ export const TicketTabNav_PurchaseTabScreen = ({navigation}: Props) => {
         });
         return;
       }
-    } else {
-      if (
-        fareProductTypeConfig.configuration.requiresTokenOnMobile &&
-        !hasInspectableMobileToken
```
Comment (rosvik) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx:78]: Pseudocode:

```
fareProductTypeConfig.configuration.requiresTokenOnMobile &&
!token.fallback && 
token.type !== "mobile"
```

**Comment by jorelosorio** [comment]
> I think this was the only use case for this screen, so unless there is a plan to use it for something else, we should delete the screen as well.

We keep the same view now but adding an extra case to show it, so now it is only shown when not fallback.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx [hunk 2]
```diff
@@ -72,25 +71,24 @@ export const TicketTabNav_PurchaseTabScreen = ({navigation}: Props) => {
...
-            nextScreen: {
-              screen: 'Root_PurchaseOverviewScreen',
-              params: {
-                fareProductTypeConfig,
-                mode: 'Ticket',
-              },
+    } else if (
+      fareProductTypeConfig.configuration.requiresTokenOnMobile &&
+      !hasInspectableMobileToken &&
+      !isFallbackState
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx:77]: Should it also check other MobileTokenStatus states here?

**Review by marius-at-atb — APPROVED** [review]

**Review by rosvik — APPROVED** [review]

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx (hunk 2 above)
Comment (gorandalum) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx]: Not completely sure about this. I saw it is now added an additional check regarding if mobile token status is `error`, but in that case it means that fallback isn't enabled. 

@jorelosorio At least add this situation to the acceptance criteria so it is up in the light, and notify Sebastian about the acceptance criteria change, to see if he agrees.

**Pull request merged**